fvm/transactionInvoker.go

```diff
@@ -75,8 +75,27 @@ func (i *TransactionInvoker) Process(
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
 		}
+
 		if mergeError := parentState.MergeState(childState, sth.EnforceInteractionLimits()); mergeError != nil {
-			processErr = fmt.Errorf("transaction invocation failed when merging state: %w", mergeError)
+			if processErr != nil {
```

Comment:
I don't quite get who creates this processErr?

Seems like it's always nil, no?

Reply:
https://github.com/onflow/flow-go/blob/90f26794b181eee66de76d5923b553a808dbcfef/fvm/transactionInvoker.go#L228

Reply:
Thanks for pointing out it, @bluesign. I see how it works now. 

I'm a big concern that this type of logic flow. 

It's better to avoid using defer here, we can do this instead:

```
function (i *TransactionInvoker) Process(...) error {
  err := i.process(...)
  if err != nil {
    ...
    sth.SetActiveState(parentState)
  }
}
```

Reply:
@zhangchiqing I don't know this area of code much, but only option I feel makes sense is if we handle some panic not here but on upper layer but also want to merge state here in any case.